examples/post-processing/rs-post-processing.cpp

```diff
@@ -0,0 +1,411 @@
+        rs2::frame colored_depth = color_map(depth_frame);
+        rs2::frame colored_filtered = color_map(filtered);
+        
+        // Tell each pointcloud object to map to to the colorized frame
+        pc.map_to(colored_depth);
+        pc_filtered.map_to(colored_filtered);
+
+        // Upload the colorized frames to OpenGL
+        app_state.tex.upload(colored_depth);
+        app_state_filtered.tex.upload(colored_filtered);
+
+        // Draw the pointclouds of the original and the filtered frames    
+        draw_pointcloud({ 0, h / 2, w / 2, h / 2 }, app_state, depth_points);
+        draw_pointcloud({ w / 2, h / 2, w / 2, h / 2 }, app_state_filtered, filtered_points);
+
+        // Update time of current frame's arrival
+        curr = std::chrono::high_resolution_clock::now().time_since_epoch().count();
+
+        /* In order to calibrate the velocity of the rotation to the actual displaying speed, rotate
+        pointcloud only when enough time has passed between frames */
+        if (curr - last_time > ROTATION_DELTA)
+        {
+            if (std::abs(app_state_filtered.yaw) > MAX_ANGLE)
+                velocity = -velocity;
+            app_state.yaw += velocity;
+            app_state_filtered.yaw += velocity;
+            last_time = curr;
+        }
+    }
+
+    return EXIT_SUCCESS;
+}
+catch (const rs2::error & e)
+{
+    std::cerr << "RealSense error calling " << e.get_failed_function() << "(" << e.get_failed_args() << "):\n    " << e.what() << std::endl;
+    return EXIT_FAILURE;
+}
+catch (const std::exception& e)
+{
+    std::cerr << e.what() << std::endl;
+    return EXIT_FAILURE;
+}
+
+
+/* Get the range of each filter option, which contains minimum, maximum and
+default values for that option, as well as the step */
+void get_filter_ranges(filter_ranges& ranges, rs2::decimation_filter& dec_filter, 
+        rs2::spatial_filter& spat_filter, rs2::temporal_filter& temp_filter) {
+    ranges.dec_magnitude_range = dec_filter.get_option_range(RS2_OPTION_FILTER_MAGNITUDE);
+    ranges.spat_magnitude_range = spat_filter.get_option_range(RS2_OPTION_FILTER_MAGNITUDE);
+    ranges.spat_smooth_alpha_range = spat_filter.get_option_range(RS2_OPTION_FILTER_SMOOTH_ALPHA);
+    ranges.spat_smooth_delta_range = spat_filter.get_option_range(RS2_OPTION_FILTER_SMOOTH_DELTA);
+    ranges.temp_smooth_alpha_range = temp_filter.get_option_range(RS2_OPTION_FILTER_SMOOTH_ALPHA);
+    ranges.temp_smooth_delta_range = temp_filter.get_option_range(RS2_OPTION_FILTER_SMOOTH_DELTA);
+    ranges.temp_magnitude_range = temp_filter.get_option_range(RS2_OPTION_FILTER_MAGNITUDE);
+}
+
+void set_defaults(params& params, filter_ranges& ranges) {
+    // Enable all filters by default
+    params.do_decimate = true;
+    params.do_temporal = true;
+    params.do_spatial = true;
+    params.do_disparity = true;
+
+    // Set default values for all filters
+    params.decimate_magnitude = ranges.dec_magnitude_range.def;
```

Comment:
Set defaults seems to be completely redundant - the processing block already starts with options at default values